### PR TITLE
[WIP] First attempt for reaching json schema support

### PR DESCRIPTION
**Pull request by realvictorprm**
This is work in progress, I hope it's the right direction...

### CODE REVIEW

**Comment by realvictorprm** [comment]
@tpetricek is this the right direction, what do you think?

File: src/Json/JsonSchemaValidator.fs [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+and validateEnum value : JsonValue [] option =
+    match value with
+    | JsonValue.Array content -> Some content
+    | _ -> None
+
+and validateConst value =
+
+    None
+
+and validateType value : JsonValue [] option =
```
Comment (realvictorprm) [inline src/Json/JsonSchemaValidator.fs]: Consider adding a `JsonValueType` record which concentrates focus on the type, not the values.

**Comment by realvictorprm** [comment]
My current plan is:
1. Use the existing parser for retrieving all json data.
2. Validate the json-schema-data
3. Create a json-schema record from the data
4. Generate an inferred type from the json-schema

**Comment by Thorium** [comment]
Consider adding [WIP] to title as long as this is not ok to merge...

**Comment by ovatsus** [comment]
#677

**Comment by realvictorprm** [comment]
I've overseen this comment, sorry. Will do

**Comment by realvictorprm** [comment]
Oh something is happening and I noticed that this got out of my sight. Sorry :(

**Comment by ovatsus** [comment]
No worries, yes, this is still super early. I'll be working on getting the xsd one in, have a look at that one as an example of what we would expect

**Comment by dsyme** [comment]
Closing this out as the repo has diverged a long way since the proposal was made

**Pull request closed without merging**